### PR TITLE
Fix SDPA tests

### PR DESCRIPTION
**Pull request by fxmarty**
@ydshieh testing on a T4 (& cpu), all tests pass now. Most of the failing ones were due to the fact that we run the CI on a T4 GPU, that does not support bf16.

### CODE REVIEW

File: tests/test_modeling_common.py [hunk 1]
```diff
@@ -3385,6 +3385,9 @@ def test_eager_matches_sdpa_inference(self, torch_dtype: str):
         if torch_device == "cpu" and torch_dtype == "float16":
             self.skipTest("float16 not supported on cpu")
 
+        if torch_device == "cuda" and torch_dtype == "bfloat16" and not torch.cuda.is_bf16_supported():
+            self.skipTest("the gpu used does not support bfloat16 data type")
```
Comment (amyeroberts) [inline tests/test_modeling_common.py:3389]: There have been recent efforts to make transformers tests device agnostic - #25654. 

In this case, we want to use something like `is_torch_bf16_available_on_device` here to check and skip here

**Comment by fxmarty** [comment]
Thank you @amyeroberts

File: tests/test_modeling_common.py [hunk 2]
```diff
@@ -3385,6 +3386,9 @@ def test_eager_matches_sdpa_inference(self, torch_dtype: str):
         if torch_device == "cpu" and torch_dtype == "float16":
             self.skipTest("float16 not supported on cpu")
 
+        if torch_dtype == "bfloat16" and not is_torch_bf16_available_on_device(torch_device):
```
Comment (amyeroberts) [inline tests/test_modeling_common.py:3389]: It would be great to update the check above too to use `is_torch_fp16_available_on_device`

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!

File: tests/test_modeling_common.py [hunk 3]
```diff
@@ -3382,8 +3384,13 @@ def test_eager_matches_sdpa_inference(self, torch_dtype: str):
         if not self.all_model_classes[0]._supports_sdpa:
             self.skipTest(f"{self.all_model_classes[0].__name__} does not support SDPA")
 
-        if torch_device == "cpu" and torch_dtype == "float16":
-            self.skipTest("float16 not supported on cpu")
+        if torch_dtype == "float16" and is_torch_fp16_available_on_device(torch_device):
```
Comment (ydshieh) [inline tests/test_modeling_common.py:3387]: missing `not` ?

**Review by ydshieh — APPROVED** [review]
thanks!

**Pull request merged**